### PR TITLE
Fix touch parse date error

### PR DESCRIPTION
**Pull request by deantvv**
Now it can parse `touch -d 2000-01-23 file`

Related to #2311

### CODE REVIEW

File: src/uu/touch/src/touch.rs [hunk 1]
```diff
@@ -238,10 +238,14 @@ fn parse_date(str: &str) -> FileTime {
     // be any simple specification for what format this parameter allows and I'm
     // not about to implement GNU parse_datetime.
     // http://git.savannah.gnu.org/gitweb/?p=gnulib.git;a=blob_plain;f=lib/parse-datetime.y
-    match time::strptime(str, "%c") {
-        Ok(tm) => local_tm_to_filetime(to_local(tm)),
-        Err(e) => panic!("Unable to parse date\n{}", e),
+    let formats = vec!["%c", "%F"];
+    for f in formats {
+        match time::strptime(str, f) {
```
Comment (sylvestre) [inline src/uu/touch/src/touch.rs:243]: could you please fix the clippy warning

File: src/uu/touch/src/touch.rs [hunk 2]
```diff
@@ -238,10 +238,14 @@ fn parse_date(str: &str) -> FileTime {
...
+            Ok(tm) => return local_tm_to_filetime(to_local(tm)),
+            _ => (),
+        }
     }
+    panic!("Unable to parse date\n");
```
Comment (sylvestre) [inline src/uu/touch/src/touch.rs:248]: please add a test to trigger this error (and probably use show_error instead of a panic)

**Review by sylvestre — COMMENTED** [review]
Looks great, just some minor changes

File: src/uu/touch/src/touch.rs (hunk 1 above)
Comment (deantvv) [reply src/uu/touch/src/touch.rs]: Fixed.

File: src/uu/touch/src/touch.rs (hunk 2 above)
Comment (deantvv) [reply src/uu/touch/src/touch.rs]: Test added and use `show_error`.

**Comment by sylvestre** [comment]
Increased the number of gnu tests by 2, well done :)

**Pull request merged**